src/components/DataTable/DataTable.tsx

```diff
@@ -330,6 +328,30 @@ class DataTableInner extends React.PureComponent<
     );
   };
 
+  private totalsRowHeading = () => {
+    const {
+      polaris: {intl},
+      totals,
+      totalsMarkupLabel,
+    } = this.props;
+
+    if (!totals) {
+      return intl.translate('Polaris.DataTable.totalsRowHeading');
+    }
+
+    const totalsIsPlural = totals.length > 2;
+
+    if (totalsMarkupLabel) {
+      return totalsIsPlural
+        ? totalsMarkupLabel.plural
+        : totalsMarkupLabel.singular;
+    }
+
+    return totalsIsPlural
+      ? intl.translate('Polaris.DataTable.totalsRowHeading')
+      : intl.translate('Polaris.DataTable.totalRowHeading');
+  };
```

Comment:
We could probably simplify this. What do you think of something like this?

```jsx
  private totalsRowHeading = () => {
    const {
      polaris: {intl},
      totals,
      totalsMarkupLabel,
    } = this.props;

    const totalsLabel = totalsMarkupLabel
      ? totalsMarkupLabel
      : {singular: intl.translate('Polaris.DataTable.totalsRow.singular), plural: intl.translate('Polaris.DataTable.totalsRow.singular)}


    return (!totals || totals.length < 2) ? totalsLabel.singular : totalsLabel.plural
  }
```

Reply:
@AndrewMusgrave it definitely makes things more concise, I had to change it a bit to match @chloerice earlier suggestions, and it would then look like:

```
  private totalsRowHeading = () => {
    const {
      polaris: {intl},
      totals = [],
      totalsName,
    } = this.props;

    const totalsLabel = totalsName
      ? totalsName
      : {
          singular: intl.translate('Polaris.DataTable.totalRowHeading'),
          plural: intl.translate('Polaris.DataTable.totalsRowHeading'),
        };

    return totals.filter((total) => total !== '').length > 1
      ? totalsLabel.plural
      : totalsLabel.singular;
  };
```

WDYT of that?

Reply:
Looks great to me 😄